scipy/stats/tests/test_distributions.py

```diff
@@ -1177,13 +1177,25 @@ def test_isf(self, q, ref):
 
     @pytest.mark.parametrize("rvs_loc", [1e-5, 1e10])
     @pytest.mark.parametrize("rvs_scale", [1e-2, 100, 1e8])
-    def test_fit_MLE_comp_optimizer(self, rvs_loc, rvs_scale):
+    @pytest.mark.parametrize('fix_loc', [True, False])
+    @pytest.mark.parametrize('fix_scale', [True, False])
+    def test_fit_MLE_comp_optimizer(self, rvs_loc, rvs_scale,
+                                    fix_loc, fix_scale):
+
+        if fix_scale and fix_loc:
+            pytest.skip("Nothing to fit.")
```

Comment:
I suspect that the `fit` override returns `floc` and `fscale` as the parameters in this case, whereas `superfit` raises an error. So this is a bit inconsistent (like a few other `fit` overrides, I'll bet). I don't mind much, but it's worth mentioning. 

I vaguely recall a discussion about the `ValueError: All parameters fixed. There is nothing to optimize.` message a few years ago. I think the argument for `raise`ing was that the user probably didn't realize they were fixing all the parameters, so we're letting them know rather than returning the obvious thing. I'm not sure that's the way I'd go today, so I'll let you decide.

Either way, I'd prefer that it be tested. If you decide that it will return `floc` and `fscale`, you can't just use `_assert_less_or_close_loglike`, so I'd prefer something like the suggestion below.

Reply:
Well, and this would need to be removed for the other to have an effect. I'll test locally tomorrow, and if it passes I'll push it, wait for tests to pass, and merge. If you beat me to it, great!

Reply:
In `_check_fit_input_parameters`, an error message is thrown for this case: [here](https://github.com/scipy/scipy/blob/main/scipy/stats/_continuous_distns.py#L5596). I adapted the test to catch the error message. I am also not the biggest fan of this but I did not want to change the behaviour just for one distribution here.

In my opinion fixing all fittable parameters should throw a RuntimeWarning rather than an error. If someone fits many different datasets based on some conditions, this case can occur and would do no harm for example. Part of the general distribution infrastructure rework?

Reply:
> Part of the general distribution infrastructure rework?

Yes. I don't think I'll raise an error. (Personally, I'm not inclined to emit a warning, either, but that can be decided later.)